### PR TITLE
Polish Helm chart

### PR DESCRIPTION
**Pull request by sonnysideup**
- Condenses Helm chart (i.e. removes buildkit subchart)
- Adds network policies
- Adds pod security policies
- Tests with Istio (w/ and w/o CNI plugin) v1.13.2
- Executes vector container as non-root user
- Converts controller container back to non-root user

### CODE REVIEW

File: deployments/helm/hephaestus/templates/_helpers.tpl [hunk 1]
```diff
@@ -36,35 +45,60 @@ Return the service account name.
...
 {{- end }}
 
 {{/*
-Returns the logfile directory.
+Return whether or not securityContext.seccompProfile field is supported.
 */}}
-{{- define "hephaestus.logfileDir" -}}
-/var/log/hephaestus
+{{- define "hephaestus.supportsSeccompGA" -}}
+{{- semverCompare ">1.19-0" .Capabilities.KubeVersion.Version }}
```
Comment (fraenkel) [inline deployments/helm/hephaestus/templates/_helpers.tpl:55]: with 5.2, we only support v1.19 and higher so its your call

File: deployments/helm/hephaestus/templates/_helpers.tpl [hunk 2]
```diff
@@ -8,21 +8,30 @@ Return the proper controller image name.
...
 
 {{/*
-Return the proper log processor image name.
+Return the controller standard labels.
 */}}
-{{- define "hephaestus.logprocessor.image" -}}
-{{- include "common.images.image" (dict "imageRoot" .Values.logProcessor.image "global" .Values.global) }}
+{{- define "hephaestus.controller.labels.standard" -}}
+{{- include "common.labels.standard" . }}
+app.kubernetes.io/component: controller
```
Comment (fraenkel) [inline deployments/helm/hephaestus/templates/_helpers.tpl:15]: we use `    {{- include "common.labels.component" "cainjector" | nindent 4 }}`

File: deployments/helm/hephaestus/templates/controller/clusterrole.yaml [hunk 3]
```diff
@@ -74,3 +74,11 @@ rules:
       - get
       - watch
       - patch
+  {{- if and (include "hephaestus.pspRequired" .) (include "hephaestus.istioWithoutCNI" .) }}
```
Comment (fraenkel) [inline deployments/helm/hephaestus/templates/controller/clusterrole.yaml:77]: why are both required?

File: deployments/helm/hephaestus/values.yaml [hunk 4]
```diff
@@ -13,14 +13,20 @@ fullnameOverride: ""
...
+
 # Istio configuration
 istio:
   # Enable support for environments with Istio installed
   enabled: false
-  # NOTE: check if this setting is necessary (i.e. run with Istio with CNI and PSPs)
   # Elevate pod execution permissions so that Istio's init container can modify
   # network settings when CNI plugin is NOT installed.
-  cniPluginInstalled: true
+  cniPluginInstalled: false
```
Comment (fraenkel) [inline deployments/helm/hephaestus/values.yaml:29]: all our other charts just use `cni: true/false`

**Review by fraenkel — CHANGES_REQUESTED** [review]

File: deployments/helm/hephaestus/templates/_helpers.tpl (hunk 1 above)
Comment (sonnysideup) [reply deployments/helm/hephaestus/templates/_helpers.tpl:55]: that's good to know. I feel fine removing this then.

File: deployments/helm/hephaestus/templates/_helpers.tpl (hunk 2 above)
Comment (sonnysideup) [reply deployments/helm/hephaestus/templates/_helpers.tpl:15]: I've been using the [bitnami common](https://github.com/bitnami/charts/tree/master/bitnami/common#labels) library chart for a while now and, unfortunately, they don't have that function available.

I _believe_ I wrote this function way back when and it does essentially the same thing.

File: deployments/helm/hephaestus/values.yaml (hunk 4 above)
Comment (sonnysideup) [reply deployments/helm/hephaestus/values.yaml:29]: I'm fine changing the field name here. 👍🏼

File: deployments/helm/hephaestus/templates/controller/clusterrole.yaml (hunk 3 above)
Comment (sonnysideup) [reply deployments/helm/hephaestus/templates/controller/clusterrole.yaml:77]: Good question, here's the answer:

You already know Istio w/o CNI means every Pod that is launched requires extra CAPs. and a root user. Hence, we need to do a little extra work whenever PSPs are enabled and configured to be _strict_.

The PSP API will be removed in k8s 1.25 so `include "hephaestus.pspRequired" .` macro will ensure that we don't render them whenever we start deploying to that version or greater. Without it, we'll get errors when trying to create resources for undefined object APIs and, in the case of RBAC objects, create superfluous rules that might confuse people

Comment (fraenkel) [reply deployments/helm/hephaestus/templates/controller/clusterrole.yaml:77]: I know why its needed with CNI, but given we have enablePSPs as a separate thing it should be an OR not an AND. PSPs are there for other things as well.
I agree the additional CAPs should be guarded by CNI but not the rest.

**Review by fraenkel — APPROVED** [review]

File: deployments/helm/hephaestus/templates/controller/clusterrole.yaml (hunk 3 above)
Comment (sonnysideup) [reply deployments/helm/hephaestus/templates/controller/clusterrole.yaml]: changed the condition.

**Pull request merged**